### PR TITLE
add tidal-hifi to flathub

### PR DESCRIPTION
**Pull request by Mastermindzh**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. [Discussion on Github about adding it, even though I am the owner](https://github.com/Mastermindzh/tidal-hifi/issues/55)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).

I do own it, but it is redirected to my other private domain (https://rickvanlieshout.com).
Everything eventually points back to me 😄 

- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:11]:
```suggestion
```

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:52]: this will not work at all. There is no network access at build time.

File: flathub.json
Comment (hfiguiere) [inline flathub.json:3]: you can remove this file, arm hasn't been supported for a couple of years already.

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:45]: you not using the icon in the upstream repository?

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:42]: these two files should be managed upstream. The lack of .desktop is already a bug as it is the standard for automatically have a desktop app available into the DE.

Neither are specific to flatpak either.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: flathub.json
Comment (Mastermindzh) [reply flathub.json:3]: Might want to review the "flathub.json" paragraph on [this page](https://github.com/flathub/flathub/wiki/App-Maintenance) then as the only reason I added the file was that specific paragraph which explicitly mentions an arm build.

File: com.mastermindzh.tidal-hifi.yml
Comment (Mastermindzh) [reply com.mastermindzh.tidal-hifi.yml:42]: The desktop file is different from the upstream one because the flatpak version requires `run.sh` whilst the upstream does not.

I don't understand why appdata.xml should be managed upstream, it belongs to the release as a flatpak.

Comment (hfiguiere) [reply com.mastermindzh.tidal-hifi.yml:42]: > The desktop file is different from the upstream one because the flatpak version requires `run.sh` whilst the upstream does not.

You can use the `desktop-file-edit` command to change that in the `post-install`

> I don't understand why appdata.xml should be managed upstream, it belongs to the release as a flatpak.

No it is NOT specfic to flatpak.

Comment (Mastermindzh) [reply com.mastermindzh.tidal-hifi.yml]: If you can provide an example or docs for `desktop-file-edit` I will be able to take a look.

As for the appdata/metainfo file, please see [The appdata guidelines](https://github.com/flathub/flathub/wiki/AppData-Guidelines).
Alternatively, if you have information that gives additional insights I'm willing to take a look there as well.

PS: I renamed appdata to metainfo as per the guidelines.

**Comment by hfiguiere** [comment]
bot, build com.mastermindzh.tidal-hifi

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [reply com.mastermindzh.tidal-hifi.yml]: > If you can provide an example or docs for `desktop-file-edit` I will be able to take a look.

I don't know more than `man desktop-file-edit` can tell you.

**Comment by hfiguiere** [comment]
The build is trying to download stuff off the internet.

**Comment by bageljrkhanofemus** [comment]
Yes, I had build issues due to some parts of the sources not getting placed correctly. I wasn't too familiar with flatpak or npm so I eventually gave up and used online builds hoping that would work on flathub.
The current issue is in cloning the electron-builder repo, which wasn't an issue when I tried to build it offline.

image
If we do clone it somehow, then it'd need to be moved to the correct directory somehow.

**Comment by bageljrkhanofemus** [comment]
Further investigation has yielded that there are two JQ commands to do just that, but for whatever reason they don't work. 
I tried to fix them by changing their target to ``$FLATPAK_BUILDER_BUILDDIR/main/package.json`` but that didn't fix it.

File: com.mastermindzh.tidal-hifi.yml
Comment (bageljrkhanofemus) [reply com.mastermindzh.tidal-hifi.yml:43]: The upstream icon is a non-standard size

File: com.mastermindzh.tidal-hifi.yml
Comment (bageljrkhanofemus) [inline com.mastermindzh.tidal-hifi.yml:48]: Add ``npm_config_offline: 'true'`` for offline builds

File: com.mastermindzh.tidal-hifi.yml
Comment (bageljrkhanofemus) [inline com.mastermindzh.tidal-hifi.yml:51]: change to npm install --offline for an offline build

Comment (Mar0xy) [reply com.mastermindzh.tidal-hifi.yml:51]: I have already opened an PR in the fork to fix all of the listed issues and actually take it from the binary instead of forcing users to build the application.

Comment (Mastermindzh) [reply com.mastermindzh.tidal-hifi.yml:51]: @Mar0xy, 

I've looked at your branch, if you make a PR I will accept.

Thanks!

**Comment by Mar0xy** [comment]
This is a PR already though?

Rick van Lieshout ***@***.***> schrieb am Di., 19. Apr. 2022,
09:30:

**Comment by barthalion** [comment]
@Mar0xy I think @Mastermindzh means that you can create a pull request to his `Mastermindzh/flathub` repository with your changes. Once your PR there is merged, the changes will appear here as well.

**Comment by Mar0xy** [comment]
> @Mar0xy I think @Mastermindzh means that you can create a pull request to his `Mastermindzh/flathub` repository with your changes. Once your PR there is merged, the changes will appear here as well.

Yea, I thought so cause like I said I already done that a few days ago [PR](https://github.com/Mastermindzh/flathub/pull/1)

**Comment by Mastermindzh** [comment]
@Mar0xy , thanks and apologies. Github didn't seem to find it necessary to notify me.
I like how we're using the extract of the appimage to build the flatpak now 😄 

@barthalion, should be good to go now!

**Comment by barthalion** [comment]
bot, build com.mastermindzh.tidal-hifi

**Comment by Mar0xy** [comment]
bot, build com.mastermindzh.tidal-hifi

File: flathub.json
Comment (Mar0xy) [reply flathub.json:2]: image

Comment (Mar0xy) [reply flathub.json:2]: Also I find it funny how selective you are with complaining about the flathub.json files in other PRs you looked at you ignored the exact same flathub file.

Comment (Mastermindzh) [reply flathub.json:2]: @Mar0xy, 

Earlier @hfiguiere was under the impression that the builds don't run on aarch64 anymore so he suggested the file to be removed. 
The [build clearly shows that it is being built on aarch64](https://flathub.org/builds/#/builders/7/builds/6760) so I agree that restoration of the file makes sense.

As I pointed out earlier the [official docs](https://github.com/flathub/flathub/wiki/App-Maintenance) also encourage the use of said file.

Comment (hfiguiere) [reply flathub.json:2]: not it's not funny. I have been complaining about x86_64 only sumbissions since forever. But if you want to pick on it, you're welcome and I will tell you why I didn't:

1. proprietary
2. wine
3. Flutter

For starter. There are others.

Here I'm noting that a MIT licensed software using a cross platform solution (Electron) that DO HAVE support for aarch64 isn't packaged on aarch64.

Comment (hfiguiere) [reply flathub.json:2]: arm != aarch64.

The former has been dropped two years ago.

My disappointement is the making of a package that is x86_64 when there doesn't seem to be a justification for it since the software could be built from source.

Comment (Mar0xy) [reply flathub.json:2]: > a package that is x86_64 **only** when there doesn't seem to be a justification for it

Actually I just tried building it for aarch64 without avail as we use castlabs electron with widevine for HiFi quality support and castlabs electron doesn't have aarch64 as there is no officially supported widevine cdm for it [Mention by a castlabs dev](https://github.com/castlabs/electron-releases/issues/114#issuecomment-968842678)

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:2]: do you need the GNOME platform?

File: com.mastermindzh.tidal-hifi.yml
Comment (Mar0xy) [reply com.mastermindzh.tidal-hifi.yml:30]: I agree I should have removed the 64x64 line but it slipped my mind :laughing:

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [reply com.mastermindzh.tidal-hifi.yml:2]: gtk3 is in the freedesktop sdk

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [reply com.mastermindzh.tidal-hifi.yml:30]: only 128x128 is required, OR SVG.

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:29]: the PNG is 512x512. Since you include it, better make it the right size instead of misreprensenting its size.

File: com.mastermindzh.tidal-hifi.yml
Comment (hfiguiere) [inline com.mastermindzh.tidal-hifi.yml:3]:
```suggestion
runtime-version: "21.08"
```

**Comment by Mar0xy** [comment]
bot, build com.mastermindzh.tidal-hifi

**Comment by barthalion** [comment]
bot, build com.mastermindzh.tidal-hifi

**Comment by Mar0xy** [comment]
Is it possible to add me as a maintainer after it's merged/created cause I assume the bot only sends an invite to the repository/fork owner.

**Comment by barthalion** [comment]
bot, build com.mastermindzh.tidal-hifi

**Comment by barthalion** [comment]
> Is it possible to add me as a maintainer after it's merged/created cause I assume the bot only sends an invite to the repository/fork owner.

We can invite multiple people, no worries.

**Pull request closed without merging**
